code/modules/food_and_drinks/food/snacks.dm

```diff
@@ -130,14 +132,19 @@
 			istype(W, /obj/item/weapon/hatchet) \
 		)
 		inaccurate = 1
-	else if(W.w_class <= WEIGHT_CLASS_SMALL && istype(src,/obj/item/weapon/reagent_containers/food/snacks/sliceable))
+	if(total_w_class > MAX_WEIGHT_CLASS && istype(src,/obj/item/weapon/reagent_containers/food/snacks/sliceable))
```

Comment:
You replaced the wrong if() statement; Leave this one as `else if(W.w_class <= WEIGHT_CLASS_SMALL && istype(src,/obj/item/weapon/reagent_containers/food/snacks/sliceable))`